starport/services/networkbuilder/verification.go

```diff
@@ -0,0 +1,104 @@
+package networkbuilder
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/starport/starport/pkg/jsondoc"
+)
+
+type GentxInfo struct {
+	ValidatorAddress string
+	SelfDelegation   sdk.Coin
+}
+
+// VerifyProposals if proposals are correct and simulate them with the current launch information
+// Correctness means checks that have to be performed off-chain
+func (b *Builder) VerifyProposals(ctx context.Context, chainID string, homeDir string, proposals []int, commandOut io.Writer) (bool, string, error) {
```

Comment:
`SimulateProposals` contains logic that generates the genesis and performs binary commands to check it. `VerifyProposals` is more generalist and perform static check on proposal as well.
If we add more static checks in the future it's better to separate both logics otherwise the simulate function becomes too long and not named on its purpose